fullstop-plugins/fullstop-kio-plugins/src/main/java/org/zalando/stups/fullstop/plugin/KioPlugin.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2015 Zalando SE
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zalando.stups.fullstop.plugin;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.cloudtrail.processinglibrary.model.CloudTrailEvent;
+import com.amazonaws.services.cloudtrail.processinglibrary.model.CloudTrailEventData;
+import com.amazonaws.services.ec2.AmazonEC2Client;
+import com.amazonaws.services.ec2.model.DescribeInstanceAttributeRequest;
+import com.amazonaws.services.ec2.model.DescribeInstanceAttributeResult;
+import com.amazonaws.util.Base64;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.zalando.stups.fullstop.aws.ClientProvider;
+import org.zalando.stups.fullstop.events.CloudtrailEventSupport;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.regex.Pattern.compile;
+import static java.util.regex.Pattern.matches;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.http.RequestEntity.get;
+import static org.springframework.web.util.UriComponentsBuilder.fromHttpUrl;
+
+/**
+ * @author mrandi
+ */
+
+@Component
+public class KioPlugin extends AbstractFullstopPlugin {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KioPlugin.class);
+
+    private static final String EC2_SOURCE_EVENTS = "ec2.amazonaws.com";
+    private static final String EVENT_NAME = "RunInstances";
+
+    private static Pattern pattern = compile("(application_id: )(.*)");
+
+    private final ClientProvider cachingClientProvider;
+
+    @Value("${fullstop.plugins.kio.url}/apps/{appId}")
+    private String kioApplicationUrl;
+
+    @Autowired
+    public KioPlugin(final ClientProvider cachingClientProvider) {
```

Comment:
public KioPlugin(final ClientProvider cachingClientProvider, Violationstore violationstore){
...
this.violationstore = violationstore;
}